### PR TITLE
Clean up GpuCollectLimitMeta and add in metrics

### PR DESCRIPTION
**Pull request by revans2**
I ran across this while trying to figure out what to do about cross join.  There were no metrics for limit so I added them in, and the code for `GpuCollectLimitExec` was not doing what we thought it was.  It was limiting based off of the number of batches, not the number of rows, and with a small change to the meta we could remove all of that code.

### CODE REVIEW

**Comment by revans2** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/limit.scala [hunk 1]
```diff
@@ -56,13 +60,20 @@ trait GpuBaseLimitExec extends LimitExec with GpuExec {
 
         override def next(): ColumnarBatch = {
           val batch = cbIter.next()
-          val result = if (batch.numRows() > remainingLimit) {
-            sliceBatch(batch)
-          } else {
-            batch
+          withResource(new NvtxWithMetrics("limit", NvtxColor.ORANGE, totalTime)) { _ =>
+            val result = if (batch.numRows() > remainingLimit) {
+              System.err.println(s"SLICE $remainingLimit/${batch.numRows()}")
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/limit.scala:65]: debug print

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/limit.scala [hunk 2]
```diff
@@ -56,13 +60,20 @@ trait GpuBaseLimitExec extends LimitExec with GpuExec {
...
+              sliceBatch(batch)
+            } else {
+              System.err.println(s"PASS $remainingLimit/${batch.numRows()}")
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/limit.scala:68]: debug print

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/limit.scala [hunk 3]
```diff
@@ -56,13 +60,20 @@ trait GpuBaseLimitExec extends LimitExec with GpuExec {
...
+              batch
+            }
+            numOutputBatches += 1
+            numOutputRows += result.numRows()
+            remainingLimit -= result.numRows()
+            System.err.println(s"RESULT $remainingLimit/${result.numRows()}")
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/limit.scala:74]: debug print

**Comment by revans2** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/limit.scala (hunk 3 above)
Comment (revans2) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/limit.scala]: sorry about that, I thought I had an off by one error when I tested the metrics with show.  Turns out it adds one to the limit in inserts.  I don't know why.  But `collect(1)` actually sets the limit to 2.  Odd

**Pull request merged**